src/java.base/share/classes/jdk/internal/reflect/NativeConstructorAccessorImpl.java

```diff
@@ -51,14 +57,23 @@ public Object newInstance(Object[] args)
         // be found from the generated bytecode.
         if (++numInvocations > ReflectionFactory.inflationThreshold()
                 && !c.getDeclaringClass().isHidden()
-                && !ReflectUtil.isVMAnonymousClass(c.getDeclaringClass())) {
-            ConstructorAccessorImpl acc = (ConstructorAccessorImpl)
-                new MethodAccessorGenerator().
-                    generateConstructor(c.getDeclaringClass(),
-                                        c.getParameterTypes(),
-                                        c.getExceptionTypes(),
-                                        c.getModifiers());
-            parent.setDelegate(acc);
+                && !ReflectUtil.isVMAnonymousClass(c.getDeclaringClass())
+                && accessorGenerated == false
```

Comment:
Should be `!accessorGenerated`.

Reply:
change to "generated == 0", as generated is int type now